cli/azd/cmd/actions/action_descriptor.go

```diff
@@ -148,6 +148,11 @@ type CommandGroupOptions struct {
 	RootLevelHelp RootLevelHelpOption
 }
 
+// EnvironmentOptions contains options for the environment flag and initialization
+type EnvironmentOptions struct {
+	Optional bool
```

Comment:
It might be nice to add a comment as to what `Optional` means here - If a command sets `Optional` to `true`, what does this mean in practice?  I think it means just "the middleware does not run when this is true and an environment is not injected"?

Reply:
Yes, you are correct.  When `Optional` is set to true then this middleware is skipped.  This is currently the case where an environment is created during the operation of the command.  Examples of this include `azd init` and `azd show`.

I will add an additional comment in the code.